### PR TITLE
Bump RuboCop, --safe-auto-correct the whole codebase

### PR DESCRIPTION
**Pull request by robhanlon22**
Totally understandable if this is not desirable, but I thought it couldn't hurt to open a PR!

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # this file is managed by dry-rb/devtools project
 
 AllCops:
-  TargetRubyVersion: 2.4
+  TargetRubyVersion: "2.7"
```
Comment (robhanlon22) [inline .rubocop.yml]: This was needed as pattern matching support was added

Comment (flash-gordon) [reply .rubocop.yml]: This file comes from devtools

File: dry-schema.gemspec [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # frozen_string_literal: true
+
 # this file is managed by dry-rb/devtools project
```
Comment (flash-gordon) [inline dry-schema.gemspec]: This file is automatically generated from project.yml

File: .rubocop.yml (hunk 1 above)
Comment (robhanlon22) [reply .rubocop.yml]: oops! thanks

File: Rakefile [hunk 3]
```diff
@@ -1,4 +1,3 @@
-#!/usr/bin/env rake
```
Comment (robhanlon22) [inline Rakefile]: Removing this seemed more sensible than giving the Rakefile execute permission

**Pull request merged**